src/luarocks/init.lua

```diff
@@ -1,6 +1,8 @@
 --- LuaRocks public programmatic API, version 3.0
 local luarocks = {}
 
+local doc = require("luarocks.doc")
+
```

Comment:
is there a reason this one goes ahead of the rest?

Reply:
I just wanted to separate the API modules from the regular ones, no particular reason besides this.

Reply:
so if you want to create this separation (and migrate stuff as a work in progress), I would start at the top with the clean part. So doing the API=related requires, followed by the explicit exporting of functions (see comment below)

Then a clear comment marker, and everything else below.

But that's how I would do it.

@hishamhm thoughts?